library/src/main/java/com/google/android/exoplayer2/extractor/mkv/MatroskaExtractor.java

```diff
@@ -1608,4 +1628,118 @@ private static boolean parseMsAcmCodecPrivate(ParsableByteArray buffer) throws P
 
   }
 
+  private class TrueHDSampleHolder {
```

Comment:
I think it's good to do the batching in MatroskaExtractor by calling sampleMetadata once every 16 samples in commitSampleToOutput. But it would be preferable to locate the start of the syncframe and get the encoded frame size in AudioTrack/Ac3Util. This all assumes that you are guaranteed to fit a syncframe in 16 samples, which I'm not sure about.